docs/platforms/python/migration/1.x-to-2.x.mdx

```diff
@@ -0,0 +1,101 @@
+
+</Note>
+
+
+If you want to move to version 2.x of the `sentry-python` SDK, we provide a short guide here of the most common patterns.
+
+While the top level API stays the same for the most part, there's a number of deprecations and breaking changes in 2.0. Here's a selection that we think might affect the most people; for the full list of changes check out the [detailed migration guide in the repository](https://github.com/getsentry/sentry-python/blob/sentry-sdk-2.0/MIGRATION_GUIDE.md).
+
+## Configuration options
+
+The `profiles_sample_rate` and `profiler_mode` options are not experimental anymore and can be used directly. Setting them via `_experiments` is deprecated and will be removed in the future.
+
+The deprecated `with_locals` and `request_bodies` options have been removed in favor of their new counterparts: `include_local_variables` and `max_request_body_size`, respectively.
+
+**Old**:
+
+<SignInNote />
+
+```python
+import sentry_sdk
+
+sentry_sdk.init(
+    dsn="___PUBLIC_DSN___",
+    with_locals=False,
+    request_bodies="always",
+    _experiments={
+        "profiles_sample_rate": 1.0,
+        "profiler_mode": "thread",
+    },
+)
+```
+
+**New**:
+
+<SignInNote />
+
+```python
+import sentry_sdk
+
+sentry_sdk.init(
+    dsn="___PUBLIC_DSN___",
+    include_local_variables=False,  # replaces with_locals
+    max_request_body_size="always",  # replaces request_bodies
+    profiles_sample_rate=1.0,  # replaces _experiments["profiles_sample_rate"]
+    profiler_mode="thread",  # replaces _experiments["profiler_mode"]
+)
+```
+
+## Python version support
+
+Sentry Python SDK 2.0 now only supports Python 3.6 and higher. If you're on Python 2.7 or 3.5 and lower, you'll need to stay on 1.x.
+
+## API
+
+The deprecated top level API function `last_event_id()` was removed. The last event ID is still returned by `capture_event()`, `capture_exception()` and `capture_message()`.
+
+## Custom instrumentation
+
+If you create a transaction manually and later mutate the transaction in a `configure_scope` block this does not work anymore. Here is a recipe on how to change your code to make it work:
+
+**Old**:
+
+<SignInNote />
+
+```python
+transaction = sentry_sdk.transaction(...)
+
+# later in the code execution:
+
+with sentry_sdk.configure_scope() as scope:
+    scope.set_transaction_name("new-transaction-name")
+```
+
+**New**:
+
+<SignInNote />
+
+```python
+transaction = sentry_sdk.transaction(...)
+
+# later in the code execution:
+
+scope = sentry_sdk.Scope.get_current_scope()
+scope.set_transaction_name("new-transaction-name")
+```
+
+## Transport
+
+Sentry Python 2.0 is now only compatible with self-hosted Sentry `v20.6.0` and above, since the SDK now sends all events to the `/envelope` endpoint. Self-hosted Sentry users must upgrade their self-hosted instance to a compatible version. No action is needed for SaaS users and self-hosted users already running a compatible Sentry version.
```

Comment:
```suggestion
Sentry Python 2.0 is now only compatible with self-hosted Sentry `v20.6.0` and above, since the SDK now sends all events to the `/envelope` API endpoint. Self-hosted Sentry users must upgrade their self-hosted instance to a compatible version. No action is needed for SaaS users and self-hosted users already running a compatible Sentry version.
```